imap_processing/cdf/utils.py

```diff
@@ -135,12 +135,22 @@ def write_cdf(
         ]
 
     # Convert the xarray object to a CDF
-    xarray_to_cdf(
-        dataset,
-        str(file_path),
-        terminate_on_warning=True,
-        **extra_cdf_kwargs,
-    )  # Terminate if not ISTP compliant
+    if "l2" in data_level:
```

Comment:
I'd invert this and only do something different for l1. We also want to leave l3 alone too.

Additionally, rather than two separate calls can you update the dictionaries being passed to xarray_to_cdf. This will allow you to keep the tests we have and add the keyword `istp=True` to the write_cdf() call. I hesitate to remove tests.

```python
if "l1" in data_level:
    if "istp" not in extra_cdf_kwargs:
        extra_cdf_kwargs["istp"] = False
else:
    if "terminate_on_warning" not in extra_cdf_kwargs:
        extra_cdf_kwargs["terminate_on_warning"] = True
    if "istp" not in extra_cdf_kwargs:
        extra_cdf_kwargs["istp"] = True

xarray_to_cdf(dataset, str(file_path), **extra_cdf_kwargs)
```